lib/oauth2client.js

```diff
@@ -281,6 +388,64 @@ OAuth2Client.prototype._handleGetTokenResponse = function(response, body, callba
   callback(null, tokenResponse);
 };
 
+OAuth2Client.prototype._handleGetDeviceCodeResponse = function(response, body, callback) {
+   var deviceCodeResponse;
+   try {
+      deviceCodeResponse = this._validateDeviceCodeResponse(body);
+   } catch (e) {
+      this._log.error('Error validating get user code response', e);
+      callback(e);
+      return;
+   }
+
+   callback(null, deviceCodeResponse);
+};
+
+OAuth2Client.prototype._getTokenWithPolling = function (postOptions, callback) {
+    var self = this;
+    if (self._cancelPollingRequest === true) {
+        callback(null, 'polling request is cancelled');
+        return;
+    }
+    
+    request.post(postOptions, util.createRequestHandler('Get Token', this._log, callback, 
+       function (response, body) {
+          var tokenResponse;
+          try {
+             tokenResponse = self._handlePollingResponse(body);
+          } catch (e) {
+             self._log.error('Error validating get token response', e);
+             callback(null, e);
+             return;
+          }
+        
+          if (tokenResponse.hasOwnProperty(TokenResponseFields.ERROR) && tokenResponse[TokenResponseFields.ERROR] === 'authorization_pending') {
```

Comment:
I agree for comparing strings, we should use double equals. and we parse the token response as Json, and compare the value of 'error' with desired value. This is more safe way to me.